source/validate_types.cpp

```diff
@@ -186,15 +263,15 @@ ModuleLayoutSection ValidationState_t::getLayoutStage() const {
 }
 
 void ValidationState_t::progressToNextLayoutStageOrder() {
-  if (static_cast<size_t>(current_layout_stage_) <= GetModuleOrder().size()) {
+  // Guard against going past the last element(kLayoutFunctionDefinitions)
+  if (current_layout_stage_ <= kLayoutFunctionDefinitions) {
```

Comment:
Also, in past we've had problems with different compilers complaining about the signed/unsigned comparison.  This version looks pristinely good to me.